### PR TITLE
Issue 54 : Support for parameterized anonymizers

### PR DESCRIPTION
**Pull request by Pyeroh**
Implementation of solution for [issue 54](https://github.com/realrolfje/anonimatron/issues/54)

This time, I already added documentation !

### CODE REVIEW

File: src/main/java/com/rolfje/anonimatron/anonymizer/ParameterizedAnonymizer.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+import java.util.HashMap;
+import java.util.Map;
+
+import com.rolfje.anonimatron.synonyms.Synonym;
+
+/**
+ * Provides functionality for consistently anonymizing a piece of data with provided parameters.
+ * @see com.rolfje.anonimatron.anonymizer.Anonymizer
+ */
+public interface ParameterizedAnonymizer extends Anonymizer {
```
Comment (realrolfje) [inline src/main/java/com/rolfje/anonimatron/anonymizer/ParameterizedAnonymizer.java]: This class may not be needed:
If we add a default method to Anonymizer with parameters, and call the old method from that, the old Anonymizers would still compile if I understand correctly. Another benefit is that we never have to check "if instanceof".

File: src/main/java/com/rolfje/anonimatron/anonymizer/AnonymizerService.java [hunk 2]
```diff
@@ -89,7 +89,12 @@ public Synonym anonymize(Column column, Object from) {
 
 		if (synonym == null) {
             Anonymizer anonymizer = getAnonymizer(column.getType());
-            synonym = anonymizer.anonymize(from, column.getSize(), column.isShortLived());
+            if (anonymizer instanceof ParameterizedAnonymizer) {
```
Comment (realrolfje) [inline src/main/java/com/rolfje/anonimatron/anonymizer/AnonymizerService.java]: See my comment in ParameterizedAnonymizer.java

File: src/main/java/com/rolfje/anonimatron/anonymizer/ParameterizedAnonymizer.java (hunk 1 above)
Comment (Pyeroh) [reply src/main/java/com/rolfje/anonimatron/anonymizer/ParameterizedAnonymizer.java]: Absolutely. A bit overengineered here ^^'

**Comment by Pyeroh** [comment]
That should be better now

**Comment by realrolfje** [comment]
Merged it into a feature branch, I'll include this in the next release.

**Comment by Pyeroh** [comment]
Thank you !

**Pull request merged**